scripts/compose.yml

```diff
@@ -0,0 +1,24 @@
+services:
+  redis:
+    image: docker.io/bitnami/redis:6.2
+    user: "${UID}:${GID}"
+    environment:
+      - ALLOW_EMPTY_PASSWORD=false
+      - REDIS_USERNAME=default
+      - REDIS_PASSWORD=the-strong-one
+      - REDIS_DISABLE_COMMANDS=FLUSHDB,FLUSHALL
+      - REDIS_TLS_CERT_FILE=/tls/redis.crt
+      - REDIS_TLS_KEY_FILE=/tls/redis.key
+      - REDIS_TLS_CA_FILE=/tls/ca.crt
+      - REDIS_TLS_ENABLED=yes
+      - REDIS_TLS_PORT=6379
+      - REDIS_TLS_AUTH_CLIENTS=no
+    ports:
+      - '6379:6379'
+    volumes:
+      - 'redis_data:/bitnami/redis/data'
+      - ./tests/tls:/tls
+
+volumes:
+  redis_data:
+    driver: local
```

Comment:
This docker compose file could be replaced with a classic docker run command in a bash file or embeded directly in the Terraform null_resource so users do not need to install docker compose as an extra dependency:
```
docker stop vault_redis_test || true
docker rm vault_redis_test || true

docker run -d \
  -p 6379:6379 \
  -u "${UID}:${GID}" \
  -e ALLOW_EMPTY_PASSWORD=false \
  -e REDIS_USERNAME=default \
  -e REDIS_PASSWORD=the-strong-one \
  -e REDIS_DISABLE_COMMANDS=FLUSHDB,FLUSHALL \
  -e REDIS_TLS_CERT_FILE=/tls/redis.crt \
  -e REDIS_TLS_KEY_FILE=/tls/redis.key \
  -e REDIS_TLS_CA_FILE=/tls/ca.crt \
  -e REDIS_TLS_ENABLED=yes \
  -e REDIS_TLS_PORT=6379 \
  -e REDIS_TLS_AUTH_CLIENTS=no \
  -v redis_data:/bitnami/redis/data \
  -v $(pwd)/tests/tls:/tls \
  --volume-driver local \
  --name vault_redis_test \
  docker.io/bitnami/redis:6.2
```

Reply:
Good idea. I created a new bash file, so now the docker container is created using a classic run command. I deleted the compose file.